backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/75-de.metas.fresh.reports/5782210_Add_InTransit_HUs_Report_function.sql

```diff
@@ -0,0 +1,92 @@
+DROP FUNCTION IF EXISTS report.InTransit_HUs_Report(IN p_M_Locator_ID numeric,
+                                                                        IN p_AD_Language Character Varying(6))
+;
+
+CREATE OR REPLACE FUNCTION report.InTransit_HUs_Report(IN p_M_Locator_ID numeric,
+                                                                           IN p_AD_Language Character Varying(6))
+    RETURNS TABLE
+            (
+                ProductName         character varying(255),
+                p_value             character varying(40),
+                qtyEntered          Numeric,
+                UOMSymbol           Character Varying(10),
+                documentno          character varying(30),
+                bp_value            character varying(40),
+                partner_name        character varying(255),
+                order_docno         character varying(30),
+                warehouse_name      character varying(255),
+                locator_to          character varying(255),
+                production_order_no character varying(30),
+                datestartschedule   timestamp WITH TIME ZONE,
+                renderedqrcode      text,
+                m_hu_id             numeric
+            )
+AS
+$$
+SELECT COALESCE(pt.Name, p.name) AS ProductName,
+       p.value                   AS p_value,
+
+       CASE
+           WHEN ddom.qtypicked > 0 THEN ddom.qtypicked
+           ELSE ddol.qtyentered
+           END                   AS QtyEntered,
+
+       uomt.uomsymbol            AS UOMSymbol,
+       ddo.documentno            AS documentno,
+       bp.value                  AS bp_value,
+       bp.name                   AS partner_name,
+       o.documentno              AS order_docno,
+       w.name                    AS warehouse_name,
+       loc_to.value              AS locator_to,
+       pp.documentno             AS production_order_no,
+       pp.datestartschedule      AS datestartschedule,
+       qr.renderedqrcode         AS renderedqrcode,
+       hu.m_hu_id                AS m_hu_id
+
+FROM M_HU hu
+
+         -- Join to get HUs on the specified locator
+         INNER JOIN M_Locator loc ON hu.M_Locator_ID = loc.M_Locator_ID
+
+    -- Join DD Order data via DD_OrderLine_HU_Candidate (as per requirement)
+         INNER JOIN DD_OrderLine_HU_Candidate ddolhc ON ddolhc.PickFrom_HU_ID = hu.M_HU_ID
```

Comment:
i think u shall join on DD_OrderLine_HU_Candidate.M_HU_ID

Reply:
and then u could use DD_OrderLine_HU_Candidate.QtyPicked.

so you might not need DD_Order_MoveSchedule